packages/hash/frontend/src/blocks/page/BlockConfigMenu/BlockConfigMenu.tsx

```diff
@@ -0,0 +1,190 @@
+import { useKey } from "rooks";
+import { get } from "lodash";
+
+import {
+  Checkbox,
+  FormControlLabel,
+  Box,
+  Typography,
+  MenuItem,
+  Popover,
+} from "@mui/material";
+import {
+  ChangeEvent,
+  ForwardedRef,
+  useEffect,
+  useRef,
+  useState,
+  VoidFunctionComponent,
+} from "react";
+import { JSONObject, JSONValue } from "blockprotocol";
+import { BlockEntity } from "@hashintel/hash-shared/entity";
+import { JsonSchema } from "@hashintel/hash-shared/json-utils";
+import { bindPopover, PopupState } from "material-ui-popup-state/hooks";
+
+import { TextField } from "../../../shared/ui/text-field";
+
+const extractConfigPropertySchemas = (blockSchema: JsonSchema) =>
+  Object.entries(blockSchema.properties ?? {}).filter(([name]) =>
+    blockSchema.configProperties?.includes(name),
```

Comment:
I wonder if we should use `block-metadata` → `configPanel` (`controls`) → `properties` instead. Deviating from `schema.org` too far feels a bit strange. Also not sure how we can mix schema auto generation and adding this field.

Reply:
By `schema.org` I meant ‘conventional’ shapes of the schema files. When we add extra properties in block schemas, it makes this object less friendly for third-party-tools (e.g. `typescript-json-schema`)

Reply:
Right, I'm with you - you definitely mean JSON Schema, though. schema.org is an entirely separate ontology designed to mark up web pages with information describing the entities on them.